### ISSUE TITLE
issues with syncPeers

### ISSUE THREAD

**Issue by simonmorgenthaler** [LiskHQ/lisk#434]
In my test, syncPeers didn't work correctly.
I did two tests, the one with the peers in config.json, the other one with the -y flag on the rebuild command.

config.json:
the node didn't manage to sync from the configured peers. It always got a `Received invalid height from peer:` error from these peers, the sync never started

-y flag:
the node started to sync, but i queried and loaded the blocks from any peer, not the ones I submitted with the -y flag to the rebuild command

**Comment by simonmorgenthaler** [comment]
testreport: https://forum.lisk.io/viewtopic.php?f=28&t=1412&p=10451#p10451

**Comment by MaciejBaj** [comment]
Thanks for testing it!

Syncing with peers works only with parameter -y or --sync. syncPeers specified in config.json are skipped and will work as before (so using decentralized way to find the peers). I will add this feature.

I couldn't replicate the error- synchronizing with different nodes than specified with -y parameter. Can you provide more details regarding it?

**Comment by simonmorgenthaler** [comment]
hm, with the peers in syncPeers in the config.json, I saw in the logs that it tried to connect only to those configured peers. So, it seems it indeed tried to sync from them only. But sync didn't start.

I used the following command to test the `-y` flag:
`./lisk.sh rebuild -u https://testnet-snapshot.lisknode.io -y <ip_peer1>:7000,<ip_peer2>:7000`
sync started, but it synced from several different nodes, not from the ones I specified. It looked like a regular rebuild without using the -y flag

**Comment by MaciejBaj** [comment]
well, I checked and with the latest version of lisk-build (lisk.sh also had been updated https://github.com/LiskHQ/lisk-build/issues/63) syncing with peers works as expected. Please make sure that you are using updated lisk.sh script

**Comment by simonmorgenthaler** [comment]
New test with 0.6.0b2 {"success":true,"build":"v17:14:49 13/02/2017\n","commit":"","version":"0.6.0b"}:
If I add the peer to syncPeers in config.json it seems to work correctly now. The blocks are loaded only from the specified peers.
using the -y flag still doesn't work. I used again the following command:
`./lisk.sh rebuild -u https://testnet-snapshot.lisknode.io -y <ip1>:7000,<ip2>:7000`
Syncing started, but the blocks are loaded from a peer/IP that was not specified (syncPeers in config.json was empty). I verified it two times.
